crates/holochain_cascade/tests/tests/get_activity.rs

```diff
@@ -598,6 +592,32 @@ async fn test_must_get_agent_activity(
     data: Data,
     author: AgentPubKey,
     filter: ChainFilter,
+) -> MustGetAgentActivityResponse {
+    test_must_get_agent_activity_inner(data, author, filter).await
+}
+
+#[cfg(feature = "unstable-warrants")]
+#[test_case(
+    Data { dht: agent_chain(&[(0, 0..3)]), warrants: vec![warrant(1, 1)], ..Default::default() },
+    agent_hash(&[0]), ChainFilter::new(action_hash(&[1]))
+    => matches MustGetAgentActivityResponse::Activity { activity, .. } if activity.len() == 2; "1 to genesis with dht 0 till 2 with 1 unrelated chain warrant")]
+#[test_case(
+    Data { dht: agent_chain(&[(0, 0..3)]), warrants: vec![warrant(0, 0)], ..Default::default() },
+    agent_hash(&[0]), ChainFilter::new(action_hash(&[1]))
+    => matches MustGetAgentActivityResponse::Activity { warrants, .. } if warrants.len() == 1; "1 to genesis with dht 0 till 2 with 1 chain warrant")]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_must_get_agent_activity_with_warrants(
+    data: Data,
+    author: AgentPubKey,
+    filter: ChainFilter,
+) -> MustGetAgentActivityResponse {
+    test_must_get_agent_activity_inner(data, author, filter).await
+}
+
+async fn test_must_get_agent_activity_inner(
```

Comment:
Splitting out a shared function to not have to duplicate all code and also run the non-warrant tests regardless of the feature.